### ISSUE TITLE
复现tts语音合成效果不一致问题

### ISSUE THREAD

**Issue by LegendSun0** [k2-fsa/sherpa-onnx#2624]
在ARM-cpu下使用的是https://github.com/k2-fsa/sherpa-onnx/blob/master/python-api-examples/offline-tts.py脚本
下载的是https://github.com/k2-fsa/sherpa-onnx/releases/download/tts-models/sherpa-onnx-vits-zh-ll.tar.bz2模型
命令如下：
python3 offline-tts.py  --vits-model=./sherpa-onnx-vits-zh-ll/model.onnx  --vits-lexicon=./sherpa-onnx-vits-zh-ll/lexicon.txt  --vits-tokens=./sherpa-onnx-vits-zh-ll/tokens.txt  --tts-rule-fsts=./sherpa-onnx-vits-zh-ll/number.fst  --vits-dict-dir=./sherpa-onnx-vits-zh-ll/dict  --sid=4  --output-filename=./test-4-2.wav  "35年前，他出生于长沙，在长白山长大。9年前他当上了银行的领导，主管行政。"

同时也使用了sherpa-onnx-offline-tts可执行文件方式
命令如下：
sherpa-onnx-offline-tts \
  --vits-model=./sherpa-onnx-vits-zh-ll/model.onnx \
  --vits-dict-dir=./sherpa-onnx-vits-zh-ll/dict \
  --vits-lexicon=./sherpa-onnx-vits-zh-ll/lexicon.txt \
  --vits-tokens=./sherpa-onnx-vits-zh-ll/tokens.txt \
  --tts-rule-fsts=./sherpa-onnx-vits-zh-ll/number.fst \
  --sid=4 \
  --output-filename="./4-heteronym.wav" \
  "35年前，他于长沙出生, 在长白山长大。9年前他当上了银行的领导，主管行政。"

生成的语音如下：

[test-4-2.wav](https://github.com/user-attachments/files/22490522/test-4-2.wav)

我自己生成的语音对比文档提供的要快3秒，语音语速快没有停顿。

<img width="1266" height="904" alt="Image" src="https://github.com/user-attachments/assets/4a9e4915-8047-49ae-9a10-a84aac82821a" />

安装库版本：
Package                     Version
--------------------------- -----------------------
absl-py                     2.1.0
accelerate                  0.34.2
aliyun-python-sdk-core      2.16.0
aliyun-python-sdk-kms       2.16.5
annotated-types             0.7.0
antlr4-python3-runtime      4.9.3
apex                        0.1.dev20241015+ascend
ascendebug                  0.1.0
ascendie                    1.0rc3
attrs                       24.2.0
audioread                   3.0.1
auto_tune                   0.1.0
blinker                     1.9.0
Brotli                      1.1.0
certifi                     2024.8.30
cffi                        1.17.1
charset-normalizer          3.3.2
click                       8.1.7
cloudpickle                 3.0.0
colorama                    0.4.6
contourpy                   1.3.0
cpm-kernels                 1.0.11
crcmod                      1.7
cryptography                45.0.3
cycler                      0.12.1
dataflow                    0.0.1
decorator                   5.1.1
diffusers                   0.29.0
editdistance                0.8.1
et-xmlfile                  1.1.0
filelock                    3.16.1
Flask                       3.1.1
fonttools                   4.54.1
fsspec                      2024.9.0
ftfy                        6.3.1
funasr                      1.2.7
fuzzywuzzy                  0.18.0
gevent                      24.2.1
geventhttpclient            2.3.1
greenlet                    3.1.1
grpcio                      1.66.1
gunicorn                    23.0.0
hccl                        0.1.0
hccl_parser                 0.1
huggingface-hub             0.25.1
hydra-core                  1.3.2
icetk                       0.0.4
idna                        3.10
importlib_metadata          8.5.0
infer_engine                1.0rc3
itsdangerous                2.2.0
jaconv                      0.4.0
jamo                        0.4.1
jieba                       0.42.1
Jinja2                      3.1.4
jmespath                    0.10.0
joblib                      1.4.2
jsonlines                   4.0.0
kaldiio                     2.18.1
kiwisolver                  1.4.7
latex2mathml                3.77.0
lazy_loader                 0.4
librosa                     0.11.0
llm_datadist                0.0.1
llm_manager_python_api_demo 1.0rc3
llvmlite                    0.44.0
loguru                      0.7.2
lxml                        5.3.0
Markdown                    3.7
MarkupSafe                  2.1.5
matplotlib                  3.9.2
mdtex2html                  1.3.0
mies_tokenizer              0.0.1
mindie_llm                  1.0rc3
mindiebenchmark             1.0rc3
mindieclient                1.0rc3
mindiesd                    1.0rc3
mindietorch                 1.0rc3+torch2.1.0.abi0
ml_dtypes                   0.5.0
model_wrapper               0.0.1
modelscope                  1.20.1
mpmath                      1.3.0
msgpack                     1.1.0
msobjdump                   0.1.0
networkx                    3.3
nltk                        3.9.1
numba                       0.61.2
numpy                       1.26.4
omegaconf                   2.3.0
op_compile_tool             0.1.0
op_gen                      0.1
op_test_frame               0.1
opc_tool                    0.1.0
openpyxl                    3.1.5
oss2                        2.19.1
packaging                   25.0
pandas                      2.2.3
pathlib2                    2.3.7.post1
peft                        0.11.1
pillow                      10.4.0
pip                         24.0
platformdirs                4.3.8
pooch                       1.8.2
portalocker                 2.10.1
prettytable                 3.11.0
protobuf                    3.20.0
psutil                      6.0.0
pyarrow                     17.0.0
pycparser                   2.22
pycryptodome                3.23.0
pydantic                    2.9.2
pydantic_core               2.23.4
pynndescent                 0.5.13
pyparsing                   3.1.4
pypinyin                    0.55.0
python-dateutil             2.9.0.post0
python-rapidjson            1.20
pytorch-wpe                 0.0.1
pytz                        2024.2
PyYAML                      6.0.2
RapidFuzz                   3.10.0
regex                       2024.9.11
requests                    2.32.3
rouge                       1.0.1
rouge_score                 0.1.2
sacrebleu                   2.4.3
safetensors                 0.4.5
schedule_search             0.0.1
scikit-learn                1.6.1
scipy                       1.14.1
sentencepiece               0.2.0
setuptools                  65.5.0
sherpa-onnx                 1.12.13
sherpa-onnx-core            1.12.13
show_kernel_debug_data      0.1.0
six                         1.16.0
sounddevice                 0.5.2
soundfile                   0.13.1
soxr                        0.5.0.post1
sympy                       1.13.3
tabulate                    0.9.0
te                          0.4.0
tensorboard                 2.18.0
tensorboard-data-server     0.7.2
tensorboardX                2.6.2.2
termcolor                   2.4.0
thefuzz                     0.22.1
threadpoolctl               3.6.0
tiktoken                    0.7.0
tokenizers                  0.19.1
torch                       2.1.0
torch-complex               0.4.4
torch-npu                   2.1.0.post8.dev20241015
torchaudio                  2.1.0
torchvision                 0.16.0
tornado                     6.4.1
tqdm                        4.66.5
transformers                4.44.0
tritonclient                2.49.0
typing_extensions           4.12.2
tzdata                      2024.2
umap-learn                  0.5.7
urllib3                     2.2.3
wcwidth                     0.2.13
Werkzeug                    3.1.3
wheel                       0.44.0
zipp                        3.21.0
zope.event                  5.0
zope.interface              7.0.3

麻烦问下是什么原因导致的，怎么样才能达到官方的效果？

**Comment by leapar** [comment]
@LegendSun0 怎么搞定的？你弄好了？我也遇到这个问题

**Comment by LegendSun0** [comment]
> [@LegendSun0](https://github.com/LegendSun0) 怎么搞定的？你弄好了？我也遇到这个问题

使用以下两个参数可以复现出来官方效果
--vits-length-scale=1.5语速缩放因子（越大越慢，越小越快）
--tts-silence-scale=0.2静音时长缩放因子（越大越慢，越小越快）
